app/lib/actions/middleware/backend_services_check.rb

```diff
@@ -0,0 +1,38 @@
+module Actions
+  module Middleware
+    class BackendServicesCheck < Dynflow::Middleware
+      def plan(*args)
+        #To prevent the ping from happening multiple times, keep track on the initial entry action
+        parent = source_action
+        input_name = self.class.name.demodulize.underscore
+
+        if Setting[:check_services_before_actions] && !parent.input[input_name]
```

Comment:
I would preffer a non-dynamic input name, such as 'checked_backend_services' as an array, and every sub-sequent backend check adding their `services` array to those. So when pulp and candelpin checks run, `input[:checked_backend_services]` would be `[:candlepin, :candlepin_auth, :pulp, :pulp_auth]`